spec/controllers/users/sessions_controller_spec.rb

```diff
@@ -231,6 +231,19 @@
         post :create, user: { email: user.email.upcase, password: user.password }
       end
 
+      it 'caches unverified PII pending confirmation' do
+        user = create(:user, :signed_up)
+        create(
+          :profile,
+          deactivation_reason: :verification_pending,
+          user: user, pii: { ssn: '1234' }
+        )
+
+        post :create, user: { email: user.email.upcase, password: user.password }
+
+        expect(controller.user_session[:decrypted_pii]).to match '1234'
```

Comment:
i.e. the correct routing happens *after* complete 2FA, but the decrypting must happen before 2FA is complete.